src/serialisation/reader.js

```diff
@@ -115,9 +113,16 @@ class ArtworksReader {
     }
   }
 
-  readRecordBody(header, checkLast) {
+  readRecordBody(header, pointer) {
     try {
-      this.populateRecord(readRecordBody(this.view, header, checkLast));
+      if (shouldSkipRecordBody(header, pointer)) {
```

Comment:
The main problem with this approach is that all other clients have to have this work around.

It would be better to move this check into `readRecordBody`. 

However, this would mean

1. We would probably need to pass the pointer structure down in addition to / instead of the checkLast
2. We may also need, in future, to pass information about how much space is available to the record (see #92)
3. The readRecordBody would then return partial or incomplete records, which then has an impact on the rest of the code.